preprocessing/demand_processing.py

```diff
@@ -57,42 +60,76 @@ def get_time_periods(root):
             while time < end:
                 if time > (week + 1) * 24 * 7:
                     week += 1
-                    time_periods_in_week[week] = []
+                    time_periods_in_week[c, week] = tuplelist()
                 if time > (day + 1) * 24:
                     day += 1
-                if time not in time_periods:
-                    time_periods.append(time)
-                    time_periods_in_week[week].append(time)
-                    time_periods_in_day[day].append(time)
+                if time not in time_periods[c]:
+                    time_periods[c].append(time)
+                    try:
+                        time_periods_in_week[c, week].append(time)
+                    except:
+                        time_periods_in_week[c, week] = [time]
+                    time_periods_in_day[c, day].append(time)
                 time += time_step
     return [time_periods, time_periods_in_week, time_periods_in_day]
 
 
+def get_combined_time_periods(time_periods, time_periods_in_week, time_periods_in_day):
```

Comment:
This function can be overlooked. Does not work as intended and have been removed in later branches.

Reply:
Is this something you have created? If not we should give Sondre a heads up :)